src/test/java/com/google/api/codegen/gapic/testdata/nodejs/nodejs_library.baseline

```diff
@@ -519,6 +534,21 @@ function sampleGetBigBook(shelf) {
       const metadata = responses[1];
       const finalApiResponse = responses[2];
 
+      // Testing interating over map fields when both key and value are specified.
+      for (const [myKey, myValue] of Object.entries(result.mapListValueValue)) {
+        console.log(`key: ${myKey}, value: ${myValue}`);
+      }
+
+      // Testing interating over map fields when only key is specified.
+      for (const anotherKey of Object.keys(result.mapListValueValue)) {
+        console.log(`key: ${anotherKey}`);
+      }
+
+      // Testing interating over map fields when only value is specified.
+      for (const anotherValue of Object.values(result.mapListValueValue)) {
```

Comment:
Is this preferable to `for...in`?
(Just asking what our style preference is)

Reply:
Yep, I did check with Alex in the design doc. 
However, @beccasaurus `Object.entries()`, `Object.keys()` and `Object.values()` will break Node 11(?), but it will go EOL this April. Is it OK we do it this way already (it's still two months away)?

Reply:
For my own edification: is that Node11 EOL as a whole, or our support for it?
[This page](https://github.com/nodejs/Release) suggests Node 11 EOL is 2019-07 (but Node 10 EOL is in 2021??)

Reply:
Oh, it's Node 6 that does not support this feature.

I think Node 10 EOL is in 2021 because it's an LTS version?

Reply:
If I'm reading this correctly, it's supported in pretty much all Node.js versions:
https://node.green/#ES2015-misc-Object-static-methods-accept-primitives

Reply:
I vote that this is OK. We can revisit later if there are Node.js version issues. I like this thought

Although I would do `Object.values(result.mapListValueValue).forEach(anotherValue => { ... })`

Reply:
Hmm. I haven't seen `for (x of y)` usage yet.

@alexander-fenster is this the most idiomatic nowadays?

Samples today all use `forEach` but if the world is headed in the direction of `for... of` _and_ it's supported across versions of Node.js nowadays, I'm +1 `for... of`

Reply:
I personally like `for (x of y)` but it's a personal preference rather than an observation of where is the world going (I really have no idea of that :) )

Reply:
Also confirmed with another Node person, beckwith.

`for (x of y)` +1

Reply:
Sounds good!